src/components/headless/components/data/item-scroller/transition.component.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+
+interface TransitionProps {
+  children: (state: string) => React.ReactNode;
+  in: boolean;
+  timeout: number;
+  onEnter?: () => void;
+  onEntering?: () => void;
+  onEntered?: () => void;
+  onExit?: () => void;
+  onExiting?: () => void;
+  onExited?: () => void;
+}
+
+function Transition({
+  children,
+  in: inProp,
+  timeout,
+  onEnter,
+  onEntering,
+  onEntered,
+  onExit,
+  onExiting,
+  onExited,
+}: TransitionProps) {
+  const [currentState, setCurrentState] = useState(inProp ? 'entering' : 'exited');
+
+  useEffect(() => {
+    if (inProp) {
+      setCurrentState('entering');
+      onEnter && onEnter();
+      const enterTimeout = setTimeout(() => {
+        setCurrentState('entered');
+        onEntered && onEntered();
```

Comment:
Again this can be written `onEntered?.();`. Instead of commenting on each of these I'll just say you can change them all.